sshd.rb

```diff
@@ -0,0 +1,121 @@
+      <listentry>LANG LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES</listentry> 
+      <listentry>LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT</listentry> 
+      <listentry>LC_IDENTIFICATION LC_ALL</listentry> 
+    </AcceptEnv>
+    <AllowTcpForwarding config:type="list">
+      <listentry>yes</listentry> 
+    </AllowTcpForwarding>
+    <Banner config:type="list">
+      <listentry>/etc/issue</listentry> 
+    </Banner>
+    <Compression config:type="list">
+      <listentry>yes</listentry> 
+    </Compression>
+    <MaxAuthTries config:type="list">
+      <listentry>6</listentry> 
+    </MaxAuthTries>
+    <PasswordAuthentication config:type="list">
+      <listentry>yes</listentry> 
+    </PasswordAuthentication>
+    <PermitRootLogin config:type="list">
+      <listentry>no</listentry> 
+    </PermitRootLogin>
+    <PermitUserEnvironment config:type="list">
+      <listentry>yes</listentry> 
+    </PermitUserEnvironment>
+    <PrintMotd config:type="list">
+      <listentry>yes</listentry> 
+    </PrintMotd>
+    <Protocol config:type="list">
+      <listentry>2</listentry> 
+    </Protocol>
+    <PubkeyAuthentication config:type="list">
+      <listentry>yes</listentry> 
+    </PubkeyAuthentication>
+    <RSAAuthentication config:type="list">
+      <listentry>yes</listentry> 
+    </RSAAuthentication>
+    <Subsystem config:type="list">
+      <listentry>sftp /usr/lib64/ssh/sftp-server</listentry> 
+    </Subsystem>
+    <UsePAM config:type="list">
+      <listentry>yes</listentry> 
+    </UsePAM>
+    <X11Forwarding config:type="list">
+      <listentry>yes</listentry> 
+    </X11Forwarding>
+  </config>
+  <status config:type="boolean">true</status> 
+</sshd>
+EOF
+
+require "rexml/document"
+require "augeas"
+SSHD = "/etc/ssh/sshd_config"
+
+# Quick hack, it should be possible to configure REXML to accept (or ignore)
+# the config namespace for attributes
+sshd_section.gsub!("config:", "")
+xml = REXML::Document.new sshd_section
+
+# Prevent Augeas from loading all known files in the system, we only
+# care about sshd_config
+Augeas::open(nil, nil, Augeas::NO_MODL_AUTOLOAD) do |augeas|
+  augeas.transform(
+    :lens => "Sshd.lns",
+    :incl => SSHD
+  )
+  augeas.load
+
+  xml.elements.each("sshd/config/*") do |param|
+    root = "/files/#{SSHD}/#{param.name}"
+    # Position matters, we don't want our script to add stuff at the end.
+    # So search for the first occurrence in the file
+    if augeas.exists(root)
+      # Add a new node to store our values and wipe out all previous values
+      before = true
+      augeas.insert(root, param.name, before)
+      augeas.rm("#{root}[position() > 1]")
+    else
+      # Let's try at least to find the commented version. If not, it will be
+      # inserted at the end of the tree
+      comment = "/files/#{SSHD}/#comment[. =~ regexp('#{param.name} .*')]"
+      before = false
+      augeas.insert(comment, param.name, before)
+    end
```

Comment:
nice, I see we have similar ideas, compare to my set helper in config files poc https://github.com/jreidinger/config_files/blob/master/lib/config_files/grub_model.rb#L43 
It is probably needed, but then it play nicely.